nixos/lib/make-disk-image.nix

```diff
@@ -174,22 +178,31 @@ let format' = format; in let
     set -f
     sources_=(${concatStringsSep " " sources})
     targets_=(${concatStringsSep " " targets})
+    modes_=(${concatStringsSep " " modes})
     set +f
 
     for ((i = 0; i < ''${#targets_[@]}; i++)); do
       source="''${sources_[$i]}"
       target="''${targets_[$i]}"
+      mode="''${modes_[$i]}"
 
+      if [ -n "$mode" ]; then
+        rsync_chmod_flags="--chmod=$mode"
+      fi
+      # Unfortunately cptofs only supports modes, not ownership, so can't use
+      # rsync's --chown option. Instead, we change the ownerships in the
+      # VM script with chown.
+      rsync_flags="-a --no-o --no-g $rsync_chmod_flags"
```

Comment:
Hmm… I think there's a problem here, because you don't reset `rsync_chmod_flags` to `""` between two loop runs.

Reply:
right... mutability strikes again >.<

will fix.